drivers/wifi/nrf700x/CMakeLists.txt

```diff
@@ -129,14 +129,72 @@ zephyr_compile_definitions_ifdef(CONFIG_NRF700X_ON_QSPI
 )
 
 # RPU FW patch binaries based on the selected configuration
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_SYSTEM_MODE
-	-DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/default/nrf70.bin
+if(CONFIG_NRF700X_SYSTEM_MODE)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/default/nrf70.bin)
+elseif (CONFIG_NRF700X_RADIO_TEST)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/radio_test/nrf70.bin)
+elseif (CONFIG_NRF700X_SCAN_ONLY)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/scan_only/nrf70.bin)
+else()
+	# Error
+	message(FATAL_ERROR "Unsupported nRF70 patch configuration")
+endif()
+
+zephyr_compile_definitions(
+	-DCONFIG_NRF_WIFI_FW_BIN=${NRF70_PATCH}
 )
 
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_RADIO_TEST
-	-DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/radio_test/nrf70.bin
+if (CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_STORE)
```

Comment:
When PM is enabled, I guess all these steps can be skipped? Not sure how `nrf70.bin` from the filesystem gets picked up and a new `.hex` is generated and merged to `zephyr.hex` using PM.

Reply:
Correct me if I'm wrong but I think we need to do it manually regardless of the PM being enabled or not. As far as I understand PM's only responsibility is to ensure partitioning defined in pm_static.yml files (or do dynamic partitioning) and generate PM_* symbols that allow to access defined memory regions in the source code.

Reply:
Yes,  the cmake should be modified to pickup the address from PM for flashing.